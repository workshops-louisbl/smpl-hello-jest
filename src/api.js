import axios from "axios";

const url = "https://reqres.in/api/users/2";

async function getUser() {
  const { data } = await axios.get(url);

  return data;
}

export default {
  getUser,
}
