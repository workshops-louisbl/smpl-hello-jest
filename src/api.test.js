import api from "./api";

test("retrieve user from api", async () => {
  const data = await api.getUser();

  expect(data).toMatchSnapshot();
})
