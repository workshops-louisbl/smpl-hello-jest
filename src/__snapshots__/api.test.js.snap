// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`retrieve user from api 1`] = `
Object {
  "ad": Object {
    "company": "StatusCode Weekly",
    "text": "A weekly newsletter focusing on software development, infrastructure, the server, performance, and the stack end of things.",
    "url": "http://statuscode.org/",
  },
  "data": Object {
    "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/josephstein/128.jpg",
    "email": "janet.weaver@reqres.in",
    "first_name": "Janet",
    "id": 2,
    "last_name": "Weaver",
  },
}
`;
